packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js

```diff
@@ -1,50 +1,49 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { List } from '@magento/peregrine';
+import React, { useCallback } from 'react';
+import { arrayOf, bool, func, number, shape, string } from 'prop-types';
 
-import classify from 'src/classify';
+import { mergeClasses } from 'src/classify';
 import Thumbnail from './thumbnail';
 import defaultClasses from './thumbnailList.css';
 
-class ThumbnailList extends Component {
-    static propTypes = {
-        activeItemIndex: PropTypes.number,
-        classes: PropTypes.shape({
-            root: PropTypes.string
-        }),
-        items: PropTypes.arrayOf(
-            PropTypes.shape({
-                label: PropTypes.string,
-                position: PropTypes.number,
-                disabled: PropTypes.bool,
-                file: PropTypes.string.isRequired
-            })
-        ).isRequired,
-        updateActiveItemIndex: PropTypes.func.isRequired
-    };
-
-    updateActiveItemHandler = newActiveItemIndex => {
-        this.props.updateActiveItemIndex(newActiveItemIndex);
-    };
-
-    render() {
-        const { activeItemIndex, items, classes } = this.props;
-
-        return (
-            <List
```

Comment:
I replaced this with an array of mapped thumbnails to stop the re-renders that happen with `List`. Should I not have?

Reply:
The only logic used by thumbnailList is contained in the `useCarousel` hook, so I don't see any reason why this component would be needed anymore.